x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/type/DataTypesTests.java

```diff
@@ -107,6 +107,29 @@ public void testIncompatibleInterval() throws Exception {
         assertNull(compatibleInterval(INTERVAL_HOUR, INTERVAL_MONTH));
         assertNull(compatibleInterval(INTERVAL_MINUTE_TO_SECOND, INTERVAL_MONTH));
     }
+    
+    public void testEsToDataType() throws Exception {
+        String type = randomFrom("null", "boolean", "bool",
```

Comment:
Can we take the list of types to randomize from from somewhere else, so that in case we change one or we add a new one, we don't have to remember this test here to add here as well?
From `DataType` I think we can do this?

Reply:
+1

Reply:
The thing is the list is only assembled in the `DataType` itself so testing that against itself kinda defeats the purpose. And thus one ends up with a different list that needs to be maintained by hand which I see no way around hence this test in its current form.

Reply:
I've extended the list by adding all data types that are primitive to the existing list of strings, which should cover any non-complex types automatically.